tests/unit_tests/oci/test_cloud.py

```diff
@@ -290,6 +290,16 @@ def test_launch_instance(self, mock_wait_till_ready, oci_cloud, oci_mock):
         mock_network_client.list_subnets.assert_called_once()
         assert oci_cloud.get_instance.call_count == 2
 
+        # Ensure when a subnet_id is directly passed to launch
+        # no functions to obtain a subnet-id are called.
```

Comment:
shouldn't we also verify that `LaunchInstanceDetails` is called with provided `instance_id`?

Reply:
correct, amend incoming!